src/StreamJsonRpc/JsonMessageFormatter.cs

```diff
@@ -518,5 +619,93 @@ internal JsonArrayPool(ArrayPool<T> arrayPool)
 
             public void Return(T[] array) => this.arrayPool.Return(array);
         }
+
+        private class JsonProgressClientConverter : JsonConverter
+        {
+            private JsonMessageFormatter formatter;
```

Comment:
>private [](start = 12, length = 7)

tip: add `readonly` here since you don't expect to ever assign it outside the ctor.